File: P1/P1-E3/VehicleDirector.java
package EJ3;

public class VehicleDirector {
    private VehicleBuilder builder;

    public VehicleDirector(VehicleBuilder builder) {
        this.builder = builder;
    }

    public void constructVehicle() {
        builder.buildEngine(builder.vehicle.engine);
        builder.buildWheels();
        builder.buildColor(builder.vehicle.color, builder.vehicle.color_vehicle);
        builder.buildAudioSystem(builder.vehicle.audioSystem);
        builder.buildTransmission(builder.vehicle.transmission);
    }
}

File: P4/VehicleDirector.dart
// ignore_for_file: file_names, non_constant_identifier_names

import 'VehicleBuilder.dart';

class VehicleDirector {
  late VehicleBuilder builder;

  //VehicleDirector({required this.builder});
  VehicleDirector(VehicleBuilder builder){
    this.builder = builder;
  }

  void constructVehicle(){
    builder.buildEngine(builder.vehicle.engine!);
    builder.buildWheels();
    builder.buildAudioSystem(builder.vehicle.audioSystem ?? false);
    builder.buildTransmission(builder.vehicle.transmission ?? false);
  }
}
